### PR TITLE
Add command handler AddItemCommand

### PR DESCRIPTION
**Pull request by vigidio**

### CODE REVIEW

File: InventoryBoundedContext/Inventory.Domain/Commands/V1/InventoryCommand.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+namespace Zelda.BOTW.Inventory.Domain.Commands.V1
+{
+    using System;
+
+    public class InventoryCommand : ICommand
+    {
+        public int Version => 1;
+    }
+
+    public class AddItemCommand : InventoryCommand
```
Comment (karinakuroda) [inline InventoryBoundedContext/Inventory.Domain/Commands/V1/InventoryCommand.cs]: Can you move this AddItemCommand to another file ?

File: InventoryBoundedContext/Inventory.Domain/Commands/V1/InventoryCommand.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        public AddItemCommand(Guid inventoryId, Guid itemId, string itemType)
+        {
+            this.InventoryId = inventoryId;
+            this.ItemId = itemId;
+            this.ItemType = itemType;
+        }
+
+        public Guid InventoryId { get; }
+
+        public string ItemType { get; }
```
Comment (karinakuroda) [inline InventoryBoundedContext/Inventory.Domain/Commands/V1/InventoryCommand.cs]: This ItemType should be a string? What do you think about change into an enum or other class with a proper description?

File: InventoryBoundedContext/Inventory.Domain/Repository/IRepository.cs [hunk 3]
```diff
@@ -4,10 +4,15 @@ namespace Zelda.BOTW.Inventory.Domain.Repository
     using System.Threading.Tasks;
     using Zelda.BOTW.Inventory.Domain.Models.AggregateRoot;
 
-    public interface IRepository<T> where T : AggregateRoot
+    public interface IAggregateRepository<T> where T : AggregateRoot
```
Comment (karinakuroda) [inline InventoryBoundedContext/Inventory.Domain/Repository/IRepository.cs]: Can you move this IAggregateRepository to another file?

File: InventoryBoundedContext/Inventory.Domain/CommandHandlers/InventoryCommandHandler.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            this.itemRepository = itemRepository;
+        }
+
+        public async Task<IAggregateChanges> Handle(AddItemCommand message)
+        {
+            var item = await this.itemRepository.GetByIdWithSecondaryIndexAsync(message.ItemId, message.ItemType);
+
+            if (item == null)
+            {
+                throw new InvalidItemException(message.ItemId);
```
[the remark this answers is not in the document]
Comment (vigidio) [reply InventoryBoundedContext/Inventory.Domain/CommandHandlers/InventoryCommandHandler.cs]: I search for, but I could not figure out how to manage this situation in a more elegant form, I found this article, but in this case I think it is similar with the last example of the article:
http://www.codinghelmet.com/articles/why-do-we-need-guard-clauses

File: InventoryBoundedContext/Inventory.Domain/CommandHandlers/InventoryCommandHandler.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+        public InventoryCommandHandler(
+            IInventoryRepository inventoryRepository,
+            IItemRepository itemRepository)
+        {
+            this.inventoryRepository = inventoryRepository;
+            this.itemRepository = itemRepository;
+        }
+
+        public async Task<IAggregateChanges> Handle(AddItemCommand message)
```
[the remark this answers is not in the document]
Comment (vigidio) [reply InventoryBoundedContext/Inventory.Domain/CommandHandlers/InventoryCommandHandler.cs]: I am following Greg Young - SimpleCQRS example, at least in the first few commits:
https://github.com/gregoryyoung/m-r/blob/master/SimpleCQRS/CommandHandlers.cs
I will consider this in future refactorings

File: InventoryBoundedContext/Inventory.Domain/Commands/V1/InventoryCommand.cs (hunk 1 above)
Comment (vigidio) [reply InventoryBoundedContext/Inventory.Domain/Commands/V1/InventoryCommand.cs]: Same as previous answer, in the first few commits I will follow SimpleCQRS example  - by Greg Young
https://github.com/gregoryyoung/m-r/blob/master/SimpleCQRS/Commands.cs
I will consider this in future refactorings

File: InventoryBoundedContext/Inventory.Domain/Commands/V1/InventoryCommand.cs (hunk 2 above)
Comment (vigidio) [reply InventoryBoundedContext/Inventory.Domain/Commands/V1/InventoryCommand.cs]: changed as suggested, itemType enum will grow in next implementations
I was going to keep commands and events with primitive types only, but I agree and now I will only keep events with primitive types only
https://khorikov.org/posts/2019-07-31-commands-primitives/

File: InventoryBoundedContext/Inventory.Domain/Repository/IRepository.cs (hunk 3 above)
Comment (vigidio) [reply InventoryBoundedContext/Inventory.Domain/Repository/IRepository.cs]: done

File: InventoryBoundedContext/Inventory.UnitTests/InventoryCommandHandlerTests.cs [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        private readonly InventoryCommandHandler commandHandler;
+
+        [Fact]
+        public async Task GivenAddItemCommand_WhenInvalidItem_ThenShouldThrowInvalidItemException()
+        {
+            // Arrange
+            var inventoryId = Guid.NewGuid();
+            var itemId = Guid.NewGuid();
+            var itemType = ItemType.Weapon;
+            var addItemCommand = new AddItemCommand(inventoryId, itemId, itemType);
```
Comment (karinakuroda) [inline InventoryBoundedContext/Inventory.UnitTests/InventoryCommandHandlerTests.cs]: This addItemCommand seems that is missing information about the item.. or the itemType should be inside the item object 🤔

Comment (vigidio) [reply InventoryBoundedContext/Inventory.UnitTests/InventoryCommandHandlerTests.cs]: ItemId and ItemType is the composite key

**Pull request merged**